### PR TITLE
Add Temporal.Calendar.prototype.daysIn* tests.

### PR DESCRIPTION
**Pull request by FrankYFTang**
https://github.com/tc39/test262/issues/3002

@jugglinmike @ryzokuken @Ms2ger

### CODE REVIEW

**Review by rwaldron — CHANGES_REQUESTED** [review]
Please fix the directory names.

**Comment by FrankYFTang** [comment]
PTAL

**Comment by rwaldron** [comment]
@FrankYFTang how are you testing these against test262-harness?

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date.js
Comment (rwaldron) [inline test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date.js:8]: Which one?

Comment (FrankYFTang) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date.js]: sorry. Bad copy and paste.

**Comment by FrankYFTang** [comment]
PTAL

**Comment by FrankYFTang** [comment]
> @FrankYFTang how are you testing these against test262-harness?

Sorry, I do not understand your question. I prototype an implemention in v8 and code the test262 tests in v8 tree, run it and see it's outcome is reasonable or not, then I copy over the file to test262 tree and upload the PR. In my v8 build I ran

$ tools/run-tests.py --outdir=out/x64.release test262/built-ins/Temporal/Calendar*

Does this answer your question?

**Comment by rwaldron** [comment]
> I prototype an implemention in v8

This is the information I was looking for. When will any of that be available in nightlies?

**Comment by FrankYFTang** [comment]
> > I prototype an implemention in v8
> 
> This is the information I was looking for. When will any of that be available in nightlies?

I am aiming to land a rough prototype behind the flag by end of Q3 with full internationalization support. 
See https://docs.google.com/document/d/1Huu2OUlmveBh4wjgx0D7ouC9O9vSdiZWaRK3OwkQZU0/edit 
I am single hand on this project now and it is not easy to wire up ~300 public functions + 200+ AOs....

This is my main task for Q3 and so far so good.
Prototype (not landing) progress tracking https://docs.google.com/spreadsheets/d/1f-S70nmYOMbp_8aF4GE7uDCZ3caLoFnsL4-DatqMfSg/edit#gid=0

very rough prototype cl (not ready for review) in https://chromium-review.googlesource.com/c/v8/v8/+/2967755 in case you like to patch v8 and play with it. Not advise you to do so. Better wait another 2.5 months after I land most of the functionality.

**Comment by anba** [comment]
I can also upload my patches for <https://bugzilla.mozilla.org/show_bug.cgi?id=1519167>, if that is useful for anyone. I didn't yet upload them because there are still many open issues (<https://github.com/tc39/proposal-temporal/issues/1502> covers roughly half of my notes about spec issues), so I was waiting for the proposal to be a bit more polished.

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js
Comment (rwaldron) [inline test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js:34]: In the future, please make these in `assert.sameValue(actual, expected)` order. It makes a difference when the errors are reported.

**Comment by FrankYFTang** [comment]
PTAL
correct the expect/actual order. Add more tests to cover the ? AO

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js
Comment (rwaldron) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js]: In case you're wondering why this all matters, I have scripts that make nice assertion messages, but when the order of args is wrong, that script makes non-sense: 

(from Temporal/Calendar/prototype/dayOfYear/string.js)

```diff
- assert.sameValue(18, cal.dayOfYear("2019-01-18"));
- assert.sameValue(49, cal.dayOfYear("2020-02-18"));
- assert.sameValue(365, cal.dayOfYear("2019-12-31"));
- assert.sameValue(366, cal.dayOfYear("2000-12-31"));
+ assert.sameValue(
+   18,
+   cal.dayOfYear("2019-01-18"),
+   '18 must return the same value returned by cal.dayOfYear("2019-01-18")'
+ );
+ assert.sameValue(
+   49,
+   cal.dayOfYear("2020-02-18"),
+   '49 must return the same value returned by cal.dayOfYear("2020-02-18")'
+ );
+ assert.sameValue(
+   365,
+   cal.dayOfYear("2019-12-31"),
+   '365 must return the same value returned by cal.dayOfYear("2019-12-31")'
+ );
+ assert.sameValue(
+   366,
+   cal.dayOfYear("2000-12-31"),
+   '366 must return the same value returned by cal.dayOfYear("2000-12-31")'
+ );
```

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js
Comment (ptomato) [inline test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js:1]: Consider copying the `branding.js` test from https://github.com/tc39/proposal-temporal/blob/main/polyfill/test/Calendar/prototype/daysInMonth/branding.js which covers this case more comprehensively. (Same remark for the other PRs)

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js
Comment (rwaldron) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js]: And for anyone else reading along... this is why using variables for the expected value sucks...

```
assert.compareArray(actual, expected, 'The value of actual is expected to equal the value of expected');
```

Comment (rwaldron) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/plain-date-time.js]: When I swap arguments in `Temporal/Calendar/prototype/dayOfYear/string.js` and re-run the script, I get this: 

```
assert.sameValue(cal.dayOfYear("2019-01-18"), 18, 'cal.dayOfYear(2019-01-18) must return 18');
assert.sameValue(cal.dayOfYear("2020-02-18"), 49, 'cal.dayOfYear(2020-02-18) must return 49');
assert.sameValue(cal.dayOfYear("2019-12-31"), 365, 'cal.dayOfYear(2019-12-31) must return 365');
assert.sameValue(cal.dayOfYear("2000-12-31"), 366, 'cal.dayOfYear(2000-12-31) must return 366');
```

Much nicer, but I'm not thrilled that I had to manually fix all of these in every file that's already been landed—but that's my problem for missing it.

**Review by rwaldron — APPROVED** [review]

File: test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js
Comment (rwaldron) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js:1]: I've created an issue for @FrankYFTang to follow up on.

Comment (FrankYFTang) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js:1]: I would love to see temporal champion to move their test to test262 sooner than later so I could focusing on developing code on v8 instead of adding more tests. If Temporal have better tests, it will be nice if they can contribute and move them into test262 so all the browser engine integrated with test262 could take advantage of them. I prefer whoever owning those tests in temporal take charge of that instead of asking me who try to fill the gap now to do that.

Comment (ptomato) [reply test/built-ins/Temporal/Calendar/prototype/daysInMonth/throw-type-error-RequireInternalSlot.js:1]: I outlined the plan for moving them over in https://github.com/tc39/test262/issues/3002#issuecomment-887908934
Ms2ger is working on it, but is not available this week.

If I tried to cut corners and move everything over right now, the tests would start to diverge and it would be really inconvenient for ensuring that we correctly and completely converted the legacy test suite. If we accidentally dropped test coverage during this process, that would be a worse outcome in the end for engines. I realize this is inconvenient in the short term, but for this reason I'd really prefer to stick to the plan that I already outlined.

There shouldn't be a problem with moving over the `branding.js` tests since I don't expect anyone will really need to revise or change those. If you don't want to do it, that's fine, one of the maintainers can assign #3137 to me.

I also think it should be fine, when you are contributing tests, to just take a look there to check if it is already done — no need to spend any effort on duplicating tests that we will move over here ASAP anyway.

**Pull request merged**
